### ISSUE TITLE
`ReduceLROnPlateau` does not inherit `LRScheduler`

### ISSUE THREAD

**Issue by pbelcak** [pytorch/pytorch#106767]
### 📚 The doc issue

The [documentation](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html) for the `ReduceLROnPLateau` scheduler fails to mention that, unlike other LR schedulers under `torch.optim.lr_scheduler`, this class does not inherit `LRScheduler`. This leads to bugs during runtime if one simply places `ReduceLROnPLateau` scheduler instead of another `LRScheduler` sub-class and forgets to remove all `LRScheduler`-specific function calls such as `LRScheduler.get_last_lr()`.

**Side question.** Why doesn't `ReduceLROnPlateau` inherit `LRScheduler` in the first place? Its internal structure already appears to be `LRScheduler`-compatible.

### Suggest a potential alternative/fix

1. Make a clear distinction in the docs between schedulers that inherit `LRScheduler`, and those who don't.
2. Add a note to the documentation of `ReduceLROnPlateau` stating explicitly that it does not inherit `LRScheduler` and because of this does not provide the same level of service as the rest of the schedulers.

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

**Comment by pbelcak** [comment]
I did a bit of digging. Historically, `ReduceLROnPlateau` was designed to track multiple param groups while other LR schedulers were still tracking only a single LR.

This [GitHub issue](https://github.com/pytorch/pytorch/issues/2829) has argued about it in 2017, giving the `.get_lr()` returning a single real number as an argument for keeping `ReduceLROnPlateau` separate from other schedulers. 

This [PyTorch forum discussion](https://discuss.pytorch.org/t/how-to-retrieve-learning-rate-from-reducelronplateau-scheduler/54234) then cites the above GitHub issue again in 2019.

Since `.get_lr()` has been replaced by `.get_last_lr()` that can now track several param groups, I guess the above argument no longer holds, meaning that `ReduceLROnPlateau` could perhaps actually become a sub-class of `LRScheduler`. Thoughts?

**Comment by janeyx99** [comment]
Thank you for diving into this! It looks like there are several other issues very similar to this one. 

I agree with you here--having ReduceLROnPlateau inherit from LRScheduler sounds correct to me. ccing @albanD for additional historical context I may be missing.
